### PR TITLE
Solved #1 - Initialized monorepo

### PR DESCRIPTION
**Pull request by supunTE**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Editor configuration, see http://editorconfig.org
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
```
Comment (pasinducw) [inline .editorconfig:6]: this does not match with the rules set in `prettierrc.json`

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (pasinducw) [inline .eslintrc.json:1]: please find out if and why the `eslintrc.json` is needed.
- Do we need to use both prettier and eslint?
- What are the things that each of them do?

File: .husky/pre-commit-msg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+exec < /dev/tty && npx cz --hook || true
```
Comment (pasinducw) [inline .husky/pre-commit-msg]: Please add a comment, specifying what this line does

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx --no-install lint-staged
```
Comment (pasinducw) [inline .husky/pre-commit]: Please add a comment, specifying what this line does

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Add files here to ignore them from prettier formatting
+
+/dist
+/coverage
+package-lock.json
+node_modules
```
Comment (pasinducw) [inline .prettierignore]: insert newline at the end of file

File: README.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
+[![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)
+
+# Razor
```
Comment (pasinducw) [inline README.md:3]: Can we clean up this file?
- Keep the relevant commands for this project

File: apps/client-e2e/cypress.config.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { defineConfig } from 'cypress';
```
Comment (pasinducw) [inline apps/client-e2e/cypress.config.ts:1]: Please find out more about `cypress`.
No need to make any changes to this file.

File: apps/client-e2e/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (pasinducw) [inline apps/client-e2e/tsconfig.json:1]: find out what the `tsconfig` do.
No need to make any changes to this file

File: apps/client/.browserslistrc [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# This file is used by:
```
Comment (pasinducw) [inline apps/client/.browserslistrc:1]: Find out what browserslist do
No need to make any changes to this file

File: apps/client/jest.config.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
```
Comment (pasinducw) [inline apps/client/jest.config.ts:1]: Find out what `jest` do
No need to change anything here

File: apps/client/src/main.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { StrictMode } from 'react';
+import * as ReactDOM from 'react-dom/client';
+
+import App from './app/app';
+
+const root = ReactDOM.createRoot(
+	document.getElementById('root') as HTMLElement,
```
Comment (pasinducw) [inline apps/client/src/main.tsx]: don't use tabs for spacing.
let's do spaces.

File: apps/client/src/polyfills.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Polyfill stable language features. These imports will be optimized by `@babel/preset-env`.
+ *
+ * See: https://github.com/zloirock/core-js#babel
+ */
+import 'core-js/stable';
```
Comment (pasinducw) [inline apps/client/src/polyfills.ts:6]: Find out about polyfills

File: nx.json [hunk 13]
```diff
@@ -0,0 +1,84 @@
+{
+  "$schema": "./node_modules/nx/schemas/nx-schema.json",
+  "npmScope": "razor",
+  "affected": {
+    "defaultBase": "master"
```
Comment (pasinducw) [inline nx.json]: main?

File: package.json [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
+      "pre-commit-msg": "exec < /dev/tty && npx cz --hook || true"
+    }
+  },
+  "private": true,
+  "dependencies": {
+    "commitizen": "^4.2.5",
```
Comment (pasinducw) [inline package.json]: this must be a dev dependency

File: .eslintrc.json (hunk 2 above)
Comment (supunTE) [reply .eslintrc.json:1]: Decided to use only one formatter: prettier.
We don't need eslint as we debug locally and run checks on GitHub CI
related commit - 1c04b19bd37ac93a17e362f3e71f53c4bf7c2bf3

File: .husky/pre-commit (hunk 4 above)
Comment (supunTE) [reply .husky/pre-commit]: Comments added

related commit - b57fef5f442d85fd03e9ab36cc91b055d10b1ae1

File: .husky/pre-commit-msg (hunk 3 above)
Comment (supunTE) [reply .husky/pre-commit-msg]: Comments added

related commit - b57fef5f442d85fd03e9ab36cc91b055d10b1ae1

Comment (supunTE) [reply .husky/pre-commit-msg]: Comments added

related commit - b57fef5f442d85fd03e9ab36cc91b055d10b1ae1

File: .prettierignore (hunk 5 above)
Comment (supunTE) [reply .prettierignore]: newline added

related commit - a2757e71aa5bff388d5fbedca9e60fb31ed95a15

File: README.md (hunk 6 above)
Comment (supunTE) [reply README.md:3]: Readme.md cleaned

related commit - df2a47f66abcd4cec7b23e6bdd96c4fb5de83594

File: apps/client-e2e/cypress.config.ts (hunk 7 above)
Comment (supunTE) [reply apps/client-e2e/cypress.config.ts:1]: Added to things to learn list

File: apps/client-e2e/tsconfig.json (hunk 8 above)
Comment (supunTE) [reply apps/client-e2e/tsconfig.json:1]: Added to things to learn list

File: apps/client/.browserslistrc (hunk 9 above)
Comment (supunTE) [reply apps/client/.browserslistrc:1]: Added to "things to learn list"

File: apps/client/jest.config.ts (hunk 10 above)
Comment (supunTE) [reply apps/client/jest.config.ts:1]: Added to "things to learn list"

File: apps/client/src/main.tsx (hunk 11 above)
Comment (supunTE) [reply apps/client/src/main.tsx]: done

related commit - https://github.com/the-ai-team/razor/commit/1c04b19bd37ac93a17e362f3e71f53c4bf7c2bf3

File: apps/client/src/polyfills.ts (hunk 12 above)
Comment (supunTE) [reply apps/client/src/polyfills.ts:6]: Added to "things to learn list"

File: nx.json (hunk 13 above)
Comment (supunTE) [reply nx.json]: fixed

related commit - 64e9d7437f30a461f94620182579c790f8ded129

File: package.json (hunk 14 above)
Comment (supunTE) [reply package.json]: Moved

related commit - dfe257ff30a3f79da26444aeb1fa2a7bf2deb312

File: .editorconfig (hunk 1 above)
Comment (supunTE) [reply .editorconfig:6]: fixed

related commit - https://github.com/the-ai-team/razor/commit/1c04b19bd37ac93a17e362f3e71f53c4bf7c2bf3

File: .eslintrc.json (hunk 2 above)
Comment (pasinducw) [reply .eslintrc.json:1]: I don't think that ditching eslint is going to be a good idea.
See: https://prettier.io/docs/en/comparison.html

Comment (supunTE) [reply .eslintrc.json:1]: Eslint re-implemented.

related commit - 87d08ce15f1ce2ec15951cc6b82016cbd852decc

**Pull request merged**